### DISCUSSION TITLE
[v6]: How to use history listen and prevent page jump？

### DISCUSSION THREAD

**Discussion by kingller** [remix-run/react-router#8624] [Q&A] [answered]
### What is the new or updated feature that you are suggesting?

1. How to use history listen as before？ 
   When router is changed, I want to reset my scrollbar position.
2. How to prevent page jump?
   When data is changed, the router will be changed too, I want to ask user to confirm.

### Why should this feature be included?

as above

**Comment by yjq8023** [comment]
Is there a plan for the second question?

**Comment by ryanflorence** [comment, marked as answer]
Scroll restoration is now built in to React Router in v6.4 :)

https://reactrouter.com/en/main/components/scroll-restoration
